src/SUnitRange.jl

```diff
@@ -0,0 +1,12 @@
+# `Start` and `End` should be `Int`
+struct SUnitRange{Start,End} <: StaticVector{Int}
+end
+
+@pure Size(::SUnitRange{Start, End}) where {Start,End} = Size(End-Start+1)
+
+@pure @propagate_inbounds function getindex(x::SUnitRange{Start,End}, i::Int) where {Start, End}
+    @boundscheck if i < Start || i > End
```

Comment:
IdentityRanges are indeed a generalization of `:`, one that allows you to slice some or all of an axis while preserving the indices. As to how it differs from `start:stop`...if you're constructing a view `v = view(a, s)` with an AbstractVector `s`, the key identity is that `v[i] == a[s[i]]`. Since `start:stop` is a conventional `UnitRange`, it is indexed starting at 1, so `v = view(a, start:stop)` would be indexed starting at 1 no matter how `a` is indexed. In contrast, `v = view(a, IdentityRange(start:stop))` would be indexed starting with `start`, no matter how `a` is indexed.